pkg/crinit/util/util.go

```diff
@@ -170,6 +170,54 @@ func UpdateKubeconfig(pathOptions *clientcmd.PathOptions, name, endpoint,
 	return nil
 }
 
+// DeleteKubeconfigEntry helper to delete the kubeconfig file entry based on input
+// parameters.
+func DeleteKubeconfigEntry(out io.Writer, pathOptions *clientcmd.PathOptions, name,
+	kubeConfigPath string, dryRun bool) error {
+
+	pathOptions.LoadingRules.ExplicitPath = kubeConfigPath
+	kubeconfig, err := pathOptions.GetStartingConfig()
+	if err != nil {
+		return err
+	}
+
+	kubeconfigFile := pathOptions.GetDefaultFilename()
+	if pathOptions.IsExplicitFile() {
+		kubeconfigFile = pathOptions.GetExplicitFile()
+	}
+
+	if dryRun {
+		return nil
+	}
+
+	_, ok := kubeconfig.Contexts[name]
+	if !ok {
+		return fmt.Errorf("cannot delete context %s, not in %s", name, kubeconfigFile)
+	}
+
+	_, ok = kubeconfig.Clusters[name]
+	if !ok {
+		return fmt.Errorf("cannot delete cluster %s, not in %s", name, kubeconfigFile)
+	}
+
+	delete(kubeconfig.Contexts, name)
+	delete(kubeconfig.Clusters, name)
+
+	// Write the updated kubeconfig.
```

Comment:
What about the credentials?

Reply:
Thanks for the reminder! I had actually thought I should delete the user authinfo as well, but since `kubectl config` doesn't have a way to view or delete just the user authinfo, I neglected to do it when I created this function and eventually forgot to revisit it. I've added it now.